cmd/soroban-cli/src/commands/contract/deploy/wasm.rs

```diff
@@ -100,19 +109,68 @@ pub enum Error {
     Network(#[from] network::Error),
     #[error(transparent)]
     Wasm(#[from] wasm::Error),
+    #[error("alias \"{alias}\" already exist. Use --force to override it.")]
+    AliasAlreadyExist { alias: String },
 }
 
 impl Cmd {
     pub async fn run(&self) -> Result<(), Error> {
+        self.validate_alias()?;
+
         let res = self.run_against_rpc_server(None, None).await?.to_envelope();
         match res {
             TxnEnvelopeResult::TxnEnvelope(tx) => println!("{}", tx.to_xdr_base64(Limits::none())?),
             TxnEnvelopeResult::Res(contract) => {
+                self.save_contract_id(&contract);
                 println!("{contract}");
             }
         }
         Ok(())
     }
+
+    fn validate_alias(&self) -> Result<(), Error> {
+        let alias = self.alias();
+
+        if alias.is_empty() {
+            return Ok(());
+        }
+
+        let path = self.alias_path();
+
+        if path.exists() && !self.force {
+            Err(Error::AliasAlreadyExist { alias })
+        } else {
+            Ok(())
+        }
+    }
+
+    fn alias(&self) -> String {
+        match &self.alias {
+            Some(n) => n.to_string(),
+            None => String::new(),
+        }
+    }
+
+    fn alias_path(&self) -> PathBuf {
+        let config_dir = self.config.config_dir().unwrap();
+        let name = format!("{}.txt", self.alias());
+
+        config_dir.join("contract-ids").join(name)
+    }
+
+    fn save_contract_id(&self, contract: &String) {
+        let file_path = self.alias_path();
+        let dir = file_path.parent().unwrap();
+
+        let _ = create_dir_all(dir);
+
+        let mut to_file = OpenOptions::new()
+            .create(true)
+            .write(true)
+            .open(file_path)
+            .unwrap();
+        let _ = to_file.write_all(contract.as_bytes());
+    }
```

Comment:
```suggestion
    fn save_contract_id(&self, contract: &str) -> Result<(), Error> {
        let Some(file_path) = self.alias_path()? else { return Ok(()); };
        let dir = file_path.parent()?;
        create_dir_all(dir)?;
        let mut to_file = OpenOptions::new()
            .create(true)
            .write(true)
            .open(file_path)?;
        to_file.write_all(contract.as_bytes())?;
        Ok(())
    }
```
`unwrap`, while nice when prototyping (still prefer `expect` since it contains a message) does panic, which is not what we want to do.  So it's better to use a Result type or return and option so that the caller can deal with it.